### PR TITLE
[SSO] Catch SAML configuration errors in a friendlier way

### PR DESCRIPTION
**Pull request by biyeun**
## Summary
Just noticed that a brand new (but misconfigured) inactive identity provider was throwing a 500 when it shouldn't be failing that hard. This makes changes to catch SSO configuration issues and display useful error messages to the user and log actual errors only when it is absolutely necessary.

### bad redirect URL
Screen Shot 2021-06-04 at 10 35 18 AM

### any non-reply URL error for active IdP
Screen Shot 2021-06-04 at 10 38 36 AM

### bad configuration error for inactive IdP
Screen Shot 2021-06-04 at 10 32 11 AM

### missing certificate error for inactive IdP
Screen Shot 2021-06-04 at 10 34 27 AM

### a general error for an inactive IdP
Screen Shot 2021-06-04 at 10 42 07 AM

## Feature Flag
No feature flag needed. Still actively in development and not part of any primary workflows.

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage
front end not tested

### QA Plan
QA will happen when SSO work is fully complete

### Safety story
This code is not actively in use on production / still in active development.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

File: corehq/apps/sso/templates/sso/config_errors.html [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            {% if request.idp.is_active %}
+              <p class="lead">
+                {% blocktrans with request.idp.is_active as idp_name %}
+                  It looks like there is an issue with {{ idp_name }}'s configuration.
+                {% endblocktrans %}
+              </p>
+              <p>
+                {% blocktrans %}
+                  We have been alerted of this issue and are working with your
+                  account manager to resolve this problem. Thank you for your
```
Comment (millerdev) [inline corehq/apps/sso/templates/sso/config_errors.html:27]: Will this always be true? Is it possible that even though we have been alerted we missed it in the general alert noise and are not actually working on it with an account manager? If that's true that might not be the best message to send here.

**Review by millerdev — APPROVED** [review]
Will the messages in `corehq/apps/sso/templates/sso/config_errors.html` be reviewed by someone from product and/or QA?

**Review by joxl — APPROVED** [review]

**Comment by biyeun** [comment]
@millerdev yes we will review as part of UAT

File: corehq/apps/sso/templates/sso/config_errors.html (hunk 1 above)
Comment (biyeun) [reply corehq/apps/sso/templates/sso/config_errors.html]: i think you are right and i'm being too optimistic here. how about...
Screen Shot 2021-06-04 at 10 38 36 AM

**Pull request merged**